packages/ui-components/src/FormDialog.tsx

```diff
@@ -108,6 +103,27 @@ export const showFormDialog = async (
           requiredFields,
           defaultButton
         );
+      } else {
```

Comment:
Should we actually enhance the `handleSingleFieldValidation` logic instead of creating this additional one?

Reply:
@lresende good idea, I took the opportunity to refactor the entire flow now that the validation is not just done for required fields, making the code simpler. `handleSingleFieldValidation` is also enhanced as suggested and now called `isFieldValid`.